tests/myproj/CMakeLists.txt

```diff
@@ -124,6 +124,13 @@ target_include_system_directories( # Test the fix of semicolon in genex issue
 target_link_libraries(lib INTERFACE myproj_project_options myproj_project_warnings)
 target_link_system_libraries(lib INTERFACE fmt::fmt Eigen3::Eigen)
 
+option(LIB_WITH_EIGEN "Enable Eigen" OFF)
+if(LIB_WITH_EIGEN)
+  target_compile_definitions(lib INTERFACE HAS_EIGEN)
+endif()
+
+target_link_libraries(main PRIVATE lib)
+
```

Comment:
How to export this option?

Reply:
It needs to be passed from the builder of this library. project_options doesn't provide a specific mechanism for this.